### DISCUSSION TITLE
Moving forward

### DISCUSSION THREAD

**Discussion by nwunderly** [nwunderly/starlette-discord#1] [Ideas]
Hi all,

I know this lib likely doesn't have many (if any) users, but I thought I'd put this out here. I'm looking for thoughts on library design, as well as ideas for features moving forward. This is intended to be a user-friendly lib, which is why it's relatively simplistic right now.

In the future, I'd like to develop a Starlette-compatible middleware, but for now that's a pretty long way away.

One thing I'd like to do soon is clean up the data returned by Discord. We have a few options for how to format this:
1. Leave as lists, dicts, etc from parsed JSON
    - This is the current behavior. It's essentially the raw data returned by the Discord API.
2. Use discord.py data classes (or custom d.py object subclasses)
    - This would work fairly well when integrating this module with a Discord bot. There are a few drawbacks, though. They would be usable exclusively as a basic data class, since these returned objects would not be connected to a discord.py http session. This could lead to strange or confusing errors, for example when trying to use `discord.User.send()`.
3. Use custom data classes
    - These would be essentially clones of the d.py data classes,. but without any of the REST-related methods. Because of this, their behavior would be fully documented but compatibility with discord.py would still be somewhat limited.

Any feedback on the current state of the library or thoughts on what would be useful moving forward would be greatly appreciated.

\- nwunder 😎

**Comment by chrisdewa** [comment]
Hello! i just discovered your lib and i'm liking it a lot! once i'm familiarized i'll see if i can contribute in any way i can.

Thanks for making this! i know it must have taken some work and it does make it a lot easier to interact with discord.
